### DISCUSSION TITLE
Refetch every button click and get new data

### DISCUSSION THREAD

**Discussion by riky177** [TanStack/query#2335] [Q&A]
I try to Refetch every button is click and get a new data 
```js
const {refetch,data,isLoading} = useQuery(
    "getkurs",() =>fetch( `https://free.currconv.com/api/v7/convert? 
      q=${selected.country.currencyId}_IDR&compact=ultra&apiKey=${process.env.COUNTRY_API}`).then((res) => 
    res.json()),{ enabled: false, refetchOnWindowFocus: false }
  );

<button onclick={()=>refetch()}>click</button>
```

It's only work once but in second and etc it's not working , can someone help me ?

**Comment by TkDodo** [comment]
please provide a codesandbox example. clicking `refetch` will _always_ refetch.

it is however advised to not do this imperatively, but add the dependencies to your queryFn to the queryKey and let react-query do the refetching for you if the key changes.
